src/vm/threads.cpp

```diff
@@ -3060,7 +3060,12 @@ BOOL Thread::CreateNewOSThread(SIZE_T sizeToCommitOrReserve, LPTHREAD_START_ROUT
     DWORD dwCreationFlags = CREATE_SUSPENDED;
 
 #ifdef FEATURE_CORECLR
-    dwCreationFlags |=  STACK_SIZE_PARAM_IS_A_RESERVATION;
+    dwCreationFlags |= STACK_SIZE_PARAM_IS_A_RESERVATION;
+
+    if (sizeToCommitOrReserve != 0 && sizeToCommitOrReserve < THREAD_MIN_STACK_SIZE)
+    {
+        sizeToCommitOrReserve = THREAD_MIN_STACK_SIZE;
```

Comment:
I do not think that this minimum should be coupled with the amount used by EnsureSufficientExecutionStack. The idea about this minimum is just to prevent guaranteed crash when the specific stack size is too small. It should be smaller than what EnsureSufficientExecutionStack ensures . Thinking about this some more ... I am not sure whether we should have one here. It implies policy that always hurt us because of there is always somebody out there who does not like it. It may be better to just pass the value down (align it up to make the OS API happy) and crash if the value is too small.